docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+
+### Frameworks
+
+If you know that you will be using frameworks, you can add them to the manifest file.
+
+For example, if you plan to use the widget and shotgun utils frameworks for your app, add the following to the `info.yml`:
+
+```python
+# the frameworks required to run this app
+frameworks:
+    - {"name": "tk-framework-shotgunutils", "version": "v2.x.x"}
+    - {"name": "tk-framework-qtwidgets", "version": "v1.x.x"}
+```
+
+#### Minimum required framework versions
+If there is a required minimum version for a framework, the minimum_version setting can be used in the `info.yml`:
+
+```python
+# the frameworks required to run this app
+frameworks:
+    - {"name": "tk-framework-shotgunutils", "version": "v2.x.x"}
+    - {"name": "tk-framework-qtwidgets", "version": "v1.x.x", "minimum_version": "v1.5.0"}
+```
+
+The above will ensure that `v1.5.0` of `tk-framework-qtwidgets` is available for the app to use. 
+If it is not, the app will not be loaded at startup time and an error will be printed to the console. 
+When the app is updated using `tank updates`, any configured frameworks not meeting their required minimum versions will be automatically updated along with the app.
+
+For more information about frameworks and how they can be useful, check out the following links:
+
+- [The Qt Widgets Framework](https://developer.shotgunsoftware.com/tk-framework-qtwidgets/)
+- [The Shotgun utils Framework](https://developer.shotgunsoftware.com/tk-framework-shotgunutils/)
+
+### Reloading your changes
+
+If you are testing your app within software such as Maya, then as soon as you have one more dev items in your configuration, 
+Toolkit will automatically add a **Reload and Restart** option to the Shotgun menu.
+![Shotgun menu Reload and Restart option](./images/reload-restart.png)
+
+Clicking this will reload your configuration and code and then restart your engine. 
+This means that you can iterate quickly: Start Maya once, and then each time you make code or configuration changes that you want to try out, simply hit the reload and restart button and your changes will be pulled in. 
+
+{% include info title="Note" content="If you have any UIs active on screen, these will not automatically update, but you have to go in and re-launch the UIs from the menu." %}
+
+## Part 5: Testing
+When you want to test your code, you can easily invite other users to your dev sandbox by adding them to the `PipelineConfiguration` entity in Shotgun. 
+As soon as you have added a user, they will see new entries on their menus inside of Shotgun Create, and the browser actions as well as an option to pick the configuration inside of Shotgun Desktop.
+
+![Dev configuration selectable in Shotgun Desktop](./images/dev-configuration.png)
+
+{% include info title="Note" content="Make sure they have access to see your app code as well, otherwise the app will not load." %}
```

Comment:
you're just talking about read permissions on the app files/folders here?

Reply:
No, for example if you developing you might have stored your code on your local drive rather than on the server. Or may you need to copy your code locally to their machine. depends on the setup.

Reply:
ah, makes sense!